packages/plugin-relay/src/types.ts

```diff
@@ -49,20 +49,24 @@ export type RelayPluginOptions<Types extends SchemaTypes> = EmptyToOptional<{
       OutputRefShape<GlobalIDShape<Types> | string>,
       boolean,
       { id: InputFieldRef<InputShape<Types, 'ID'>> },
-      Promise<unknown>
+      Promise<Types['Objects']>
     >,
     'args' | 'resolve' | 'type'
-  >;
+  > & {
+    resolve?: (id: string, context: Types['Context']) => MaybePromise<Types['Objects']>;
+  };
   nodesQueryOptions: Omit<
     PothosSchemaTypes.QueryFieldOptions<
       Types,
       [OutputRefShape<GlobalIDShape<Types> | string>],
       FieldNullability<[unknown]>,
       { ids: InputFieldRef<InputShape<Types, 'ID'>[]> },
-      Promise<unknown>[]
+      Promise<Types['Objects']>[]
```

Comment:
I think `unknown` is the best we can do here.  It's impossible to type check this.  The type you are using there works for object types explicitly defined in builder setup, but won't work for nodes created with any of the `ref` based apis.